File: 509-fibonacci-number/509-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        if (n == 0) return 0;
        int x = 0, y = 1;
        for (; n > 1; --n) {
            y += x;
            x = y - x;
        }
        return y;
    }
};

File: 509-fibonacci-number/509-fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        if n == 0: return 0
        arr = [0] + [1] * n
        for i in range(2, n + 1):
            arr[i] = arr[i - 1] + arr[i - 2]
        return arr[n]